horace_core/sqw/page_operations/PageOp_split_sqw.m

```diff
@@ -147,29 +181,95 @@
 
             % this will remove unused  (non-contributed)  handles places
             obj.write_handles = obj.write_handles(obj.run_contributes_);
-            % Allocate resulting sqw array
+            % Allocate resulting sqw array or list of sqw files
             n_obj = numel(obj.out_img);
-            out_obj = repmat(sqw,[1,n_obj]);
+            if obj.img_filebacked_
+                out_obj = cell(1,n_obj);
+            else
+                out_obj = repmat(sqw,[1,n_obj]);
+            end
             for i=1:n_obj
                 split_img =  obj.out_img{i};
                 % Set single split object as the result of page operation
-                obj.npix          = split_img.npix;
-                obj.sig_acc_      = split_img.s;
-                obj.var_acc_      = split_img.e;
-                obj.img_          = obj.out_sqw{i}.data;
-                obj.write_handle_ = obj.write_handles{i};
-                obj.pix_          = obj.out_sqw{i}.pix;
+                if ~obj.img_filebacked_
+                    obj.npix          = split_img.npix;
+                    obj.sig_acc_      = split_img.s;
+                    obj.var_acc_      = split_img.e;
+                    obj.img_          = obj.out_sqw{i}.data;
+                end
+                obj.write_handle_   = obj.write_handles{i};
+                obj.pix_            = obj.out_sqw{i}.pix;
                 obj.pix_data_range_ = obj.obj_pix_ranges{i};
                 % finalize result as with single object and store result
                 % in the output array of objects.
-                [out_obj(i),obj] = finish_op@PageOpBase(obj,obj.out_sqw{i});
+                if obj.img_filebacked_
+                    [out_obj{i},obj] = obj.finalize_fb_obj(obj.out_sqw{i});
+                else
+                    [out_obj(i),obj] = finish_op@PageOpBase(obj,obj.out_sqw{i});
+                end
             end
         end
         %
     end
     % Class specific methods
     methods
-        function obj = prepare_split_sqw(obj,in_sqw,pix_filebacked)
+        function [out_file,obj] = finalize_fb_obj(obj,in_sqw)
+            % special overload used for filebacked operations only when
+            % source data are located in file and target data are locate in
+            % file because array of resulting filebacked objects does not
+            % fit memory.
```

Comment:
changed to 
```
            % special overload used for filebacked operations only when
            % source data are located in file and target data are located in
            % file because even the array of resulting filebacked objects
            % does not fit memory.
```
and made protected method